packages/babel-register/src/node.js

```diff
@@ -129,20 +111,27 @@ function hookExtensions(_exts) {
   });
 }
 
-hookExtensions(util.canCompile.EXTENSIONS);
+hookExtensions([".js", ".jsx", ".es6", ".es"]);
 
 export default function (opts?: Object = {}) {
-  if (opts.only != null) only = util.arrayify(opts.only, util.regexify);
-  if (opts.ignore != null) ignore = util.arrayify(opts.ignore, util.regexify);
-
-  if (opts.extensions) hookExtensions(util.arrayify(opts.extensions));
+  if (opts.extensions) hookExtensions(opts.extensions);
 
   if (opts.cache === false) cache = null;
 
   delete opts.extensions;
-  delete opts.ignore;
   delete opts.cache;
```

Comment:
never seen this one? would be nice to not nice delete everywhere. Oh and also not modify opts I suppose.. if we care about doing https://github.com/babel/babel/issues/3781